gr-utils/python/modtool/tests/test_modtool.py

```diff
@@ -0,0 +1,290 @@
+    def __init__(self, *args, **kwargs):
+        super(TestModToolCore, self).__init__(*args, **kwargs)
+        self.f_add = False
+        self.f_newmod = False
+
+    @classmethod
+    def setUpClass(cls):
+        """ create a temporary directory """
+        cls.test_dir = tempfile.mkdtemp()
+
+    @classmethod
+    def tearDownClass(cls):
+        """ remove the directory after the test """
+        shutil.rmtree(cls.test_dir)
+
+    def setUp(self):
+        """ create a new module and block before every test """
+        try:
+            warnings.simplefilter("ignore", ResourceWarning)
+            args = {'module_name':'howto', 'directory': self.test_dir}
+            ModToolNewModule(**args).run()
+        except (TypeError, ModToolException):
+            self.f_newmod = True
+        else:
+            try:
+                args = {'blockname':'square_ff', 'block_type':'general',
+                        'lang':'cpp', 'directory': self.test_dir + '/gr-howto',
+                        'add_python_qa': True}
+                ModToolAdd(**args).run()
+            except (TypeError, ModToolException):
+                self.f_add = True
+
+    def tearDown(self):
+        """ removes the created module """
+        # Required, else the new-module directory command
+        # in setup with throw exception after first test
+        ## cannot remove if directory is not created
+        if not self.f_newmod:
+            rmdir = self.test_dir + '/gr-howto'
+            shutil.rmtree(rmdir)
+
+    def test_newmod(self):
+        """ Tests for the API function newmod """
+        ## Tests for proper exceptions ##
+        test_dict = {}
+        test_dict['directory'] = self.test_dir
+        # module name not specified
+        self.assertRaises(ModToolException, ModToolNewModule(**test_dict).run)
+        test_dict['module_name'] = 'howto'
+        # expected module_name as a string instead of dict
+        self.assertRaises(TypeError, ModToolNewModule(test_dict).run)
+        # directory already exists
+        # will not be raised if the command in setup failed
+        self.assertRaises(ModToolException, ModToolNewModule(**test_dict).run)
+
+        ## Some tests for checking the created directory, sub-directories and files ##
+        test_dict['module_name'] = 'test'
+        ModToolNewModule(**test_dict).run()
+        module_dir = path.join(self.test_dir, 'gr-test')
+        self.assertTrue(path.isdir(module_dir))
+        self.assertTrue(path.isdir(path.join(module_dir, 'lib')))
+        self.assertTrue(path.isdir(path.join(module_dir, 'python')))
+        self.assertTrue(path.isdir(path.join(module_dir, 'include')))
+        self.assertTrue(path.isdir(path.join(module_dir, 'docs')))
+        self.assertTrue(path.isdir(path.join(module_dir, 'cmake')))
+        self.assertTrue(path.isdir(path.join(module_dir, 'swig')))
+        self.assertTrue(path.exists(path.join(module_dir, 'CMakeLists.txt')))
+
+        ## pylint tests ##
```

Comment:
What is the purpose of this commented section? Is it usable if uncommented?

Reply:
Yes, it is usable but for me, it is highly time-consuming.

Reply:
@primercuervo: Uncommented but now the tests may take as long as `12-13 secs`.